developer_manual/client_apis/OCS/ocs-openapi.rst

```diff
@@ -0,0 +1,329 @@
+===================
+OCS OpenAPI support
+===================
+
+This page explains you how you can add OpenAPI support to your app such that you can automatically generated an OpenAPI specification for your server code.
+
+Please read the whole tutorial before starting to adjust your app.
+
+Don't be afraid that you don't know everything from the start.
+The openapi-extractor tool gives you many warnings and fails if there is something utterly broken that would not work.
+Just let it run, it will tell you if there is something wrong.
+Psalm will also help you validate your changes.
+
+Requirements and prerequisites
+------------------------------
+
+- Psalm has been setup for the app
+- The app supports at least Nextcloud 28
+- Your APIs are exposed via OCS (see :ref:`OCS <ocscontroller>`). It is possible to also expose APIs in different ways, but those will not be explained here because they are considered legacy.
+
+Best practices
```

Comment:
I don't know how easy it is to replicate here but I really like how pest practices are communicated by the dart language (https://dart.dev/effective-dart/documentation).

They all start with a key word like `DO`, `DON'T`, `PREFER`, `AVOID` `CONSIDER` and then also give a clear example why this is good or bad.

I think that the items should at least always start with a key word as it's already done with the `Do not`

Reply:
I like this idea lot. I will try to implement something similar

Reply:
I think the best practices are a lot better now, especially with the code examples. What do you think?